kolona/task.py

```diff
@@ -83,6 +88,7 @@ async def process(self):
         """
         Process this specific task
         """
+        await asyncio.sleep(self.delay)
```

Comment:
Problem here is that this would block the execution of the worker and it would be stuck until the end of sleep and execution of the task. I would prefer if the task gets added back to the queue in this case and the worker takes the next available task. Goal is for the worker to be "stuck" for as little as necessary.

I think it should be handled similarly to how `is_ready` is. Where the worker executes `task.is_ready` and only if truthy the worker would execute the task otherwise it puts it back onto the queue and takes another task.

Reply:
Yes, you are completely right, I didn't really consider use cases where this would matter. It actually makes a lot of sense to avoid blocking the worker.

I will probably just add a check into `is_ready`.